common/djangoapps/enrollment/data.py

```diff
@@ -106,7 +106,7 @@ def create_course_enrollment(username, course_id, mode, is_active):
         CourseEnrollmentExistsError
 
     """
-    course_key = CourseKey.from_string(course_id)
+    course_key = course_key_from_string_or_404(course_id)
```

Comment:
is this an appropriate place to call this helper? is this function only called by a view?